src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -105,13 +105,13 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
             PackageDependency packageDependency = default;
             if (packageReferenceArgs.NoVersion)
             {
-                var latestVersion = await GetLatestVersionAsync(originalPackageSpec, packageReferenceArgs.PackageId, packageReferenceArgs.Logger, packageReferenceArgs.Prerelease);
+                var latestVersion = await GetLatestVersionAsync(originalPackageSpec, packageReferenceArgs.Sources, packageReferenceArgs.PackageId, packageReferenceArgs.Logger, packageReferenceArgs.Prerelease);
```

Comment:
Did you test this? :) 

Pretty sure this is not enough, as the installation is 2 step, once to get the version, once to actually download the package, and the source is only used for the version calculation.

Reply:
> Did you test this? :)
> 
> Pretty sure this is not enough, as the installation is 2 step, once to get the version, once to actually download the package, and the additional source is only used for the version calculation.

Yes, manually I did test with both local and remote private feeds.
1. With my change it get's latest version from additional user source feeds.
2. After that this additional user source feeds are passed to runner job for actual restore.

Can you refer where is `dotnet pack ....` unit tests?

Reply:
Search for PackageCommandTests...not sure why you need `Pack` in this case though.

Reply:
Added unit tests.